pkt-meta/src/dst_vpcd_lookup/mod.rs

```diff
@@ -113,9 +117,15 @@ impl VpcDiscTablesWriter {
     }
 }
 
+#[derive(Debug, Clone)]
+enum VpcDiscriminantTableValue {
+    PortRangeTable(DisjointRangesBTreeMap<PortRange, Option<VpcDiscriminant>>),
+    NoPorts(Option<VpcDiscriminant>),
```

Comment:
I think I like this enumeration approach better for the IpRangeWithPorts trait as well instead of using an empty port range to mean IPs only.  It might clean up my earlier comment about the size method as well.